File: src/main/java/cc/sukazyo/cono/morny/daemon/MornyDaemons.java
package cc.sukazyo.cono.morny.daemon;

import static cc.sukazyo.cono.morny.Log.logger;

public class MornyDaemons {
	
	public static final MedicationTimer medicationTimerInstance = new MedicationTimer();
	
	public static void start () {
		logger.info("ALL Morny Daemons starting...");
		TrackerDataManager.init();
		medicationTimerInstance.start();
		logger.info("Morny Daemons started.");
	}
	
	public static void stop () {
		
		logger.info("ALL Morny Daemons stopping...");
		
		TrackerDataManager.DAEMON.interrupt();
		medicationTimerInstance.interrupt();
		
		TrackerDataManager.trackingLock.lock();
		try { medicationTimerInstance.join(); } catch (InterruptedException e) { e.printStackTrace(System.out); }
		
		logger.info("ALL Morny Daemons STOPPED.");
		
	}
	
}

File: src/main/scala/cc/sukazyo/cono/morny/daemon/MornyDaemons.scala
package cc.sukazyo.cono.morny.daemon

import cc.sukazyo.cono.morny.Log.logger
import cc.sukazyo.cono.morny.MornyCoeur

class MornyDaemons (using val coeur: MornyCoeur) {
	
	val medicationTimer: MedicationTimer = MedicationTimer()
	val reporter: MornyReport = MornyReport()
	val eventHack: EventHacker = EventHacker()
	
	def start (): Unit = {
		
		logger notice "ALL Morny Daemons starting..."
		
		medicationTimer.start()
		reporter.start()
		
		logger notice "Morny Daemons started."
		
	}
	
	def stop (): Unit = {
		
		logger notice "stopping All Morny Daemons..."
		
		medicationTimer.stop()
		reporter.stop()
		
		logger notice "stopped ALL Morny Daemons."
	}
	
}
